sunpy/image/transform.py

```diff
@@ -90,35 +90,19 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
     displacement = np.dot(rmatrix, rot_center)
     shift = image_center - displacement
 
-    # While `use_scipy` is still supported, we have to check which method to actually use
-    method = _get_transform_method(method, use_scipy)
+    method = _get_transform_method(method)
 
     # Transform the image using the appropriate function
     rotated_image = _rotation_registry[method].function(image, rmatrix, shift, order, missing, clip)
 
     return rotated_image
 
 
-def _get_transform_method(method, use_scipy):
+def _get_transform_method(method):
     # This is re-used in affine_transform and GenericMap.rotate
     if method not in _rotation_registry:
         raise ValueError(f'Method {method} not in supported methods: '
                          f'{_rotation_registry.keys()}')
-
-    if use_scipy is not None:
-        warn_deprecated("The 'use_scipy' argument is deprecated. "
-                        "Specify the rotation method to the 'method' "
-                        "keyword argument instead.")
-        if use_scipy is True and method != 'scipy':
-            warn_user(f"Using scipy instead of {method} for rotation.")
-            method = 'scipy'
-
-    if method == 'scikit-image':
-        try:
-            import skimage  # NoQA
-        except ImportError:
-            raise ImportError("scikit-image must be installed to be usable for rotation.")
-
```

Comment:
Why move this block out of here? Isn't the point of the `_get_method` function to do validation of the choice of method all in one place?

Reply:
The rotation method for opencv check for the import inside of them and I think they should handle it.

The same goes for cupy, if it was done here, someone would need to PR it to core sunpy.

Reply:
Ah ok, thanks. That makes sense. So `_get_method` should just check the registry and return the desired method.

Reply:
That is what it does now only, before it also did the import check only for scikit image.